dev/io.openliberty.http.monitor_fat/fat/src/io/openliberty/http/monitor/fat/BaseTestClass.java

```diff
@@ -0,0 +1,193 @@
+            String line = null;
+            StringBuilder lines = new StringBuilder();
+            BufferedReader br = new BufferedReader(new InputStreamReader(con.getInputStream()));
+
+            while ((line = br.readLine()) != null && line.length() > 0) {
+                lines.append(line).append(sep);
+            }
+            Log.info(c, "requestHttpServlet", sURL);
+            return lines.toString();
+        } catch (IOException e) {
+            Log.info(c, "requestHttpServlet", "Encountered IO exception " + e);
+            return null;
+        } catch (Exception e) {
+            Log.info(c, "requestHttpServlet", "Encountered an exception " + e);
+            return null;
+        } finally {
+            if (con != null)
+                con.disconnect();
+        }
+
+    }
+
+    protected String getVendorMetrics(LibertyServer server) throws Exception {
+        String vendorMetricsOutput = requestHttpSecureServlet("/metrics?scope=vendor", server, HttpMethod.GET);
+        Log.info(c, "getVendorMetrics", vendorMetricsOutput);
+        return vendorMetricsOutput;
+    }
+
+    protected boolean validatePrometheusHTTPMetric(String vendorMetricsOutput, String route, String responseStatus, String requestMethod) {
+        return validatePrometheusHTTPMetric(vendorMetricsOutput, route, responseStatus, requestMethod, null, null);
+    }
+
+    protected boolean validatePrometheusHTTPMetricWithErrorType(String vendorMetricsOutput, String route, String responseStatus, String requestMethod, String errorType) {
+        return validatePrometheusHTTPMetric(vendorMetricsOutput, route, responseStatus, requestMethod, null, errorType);
+    }
+
+    protected boolean validatePrometheusHTTPMetric(String vendorMetricsOutput, String route, String responseStatus, String requestMethod, String count, String errorType) {
+
+        if (count == null) {
+            count = "[0-9]+\\.[0-9]+";
+        }
+
+        if (errorType == null) {
+            errorType = "";
+        }
+        String matchString = "http_server_request_duration_seconds_count\\{error_type=\"" + errorType + "\",http_route=\"" + route
+                             + "\",http_scheme=\"http\",mp_scope=\"vendor\",network_name=\"HTTP\",network_version=\"1\\.[01]\",request_method=\"" + requestMethod
+                             + "\",response_status=\"" + responseStatus + "\",server_name=\"localhost\",server_port=\"[0-9]+\",\\} " + count;
+
+        try (Scanner sc = new Scanner(vendorMetricsOutput)) {
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine();
+                /*
+                 * Skip things we don't care about for perfomance
+                 */
+                if (!line.startsWith("http_server_request_duration_seconds_count")) {
+                    continue;
+                }
+
+                if (line.matches(matchString)) {
+                    Log.info(c, "validatePrometheusHTTPMetric", "Matched With line: " + line);
```

Comment:
Is it possible to add a check for the metric value greater than `0.0`?

Reply:
Fixed in https://github.com/OpenLiberty/open-liberty/pull/28565/commits/3b17a4d12b48f93656ea09ed58209ff55e7c6843